Shared/Samples/Identify raster cell/IdentifyRasterCellView.swift

```diff
@@ -0,0 +1,252 @@
+        }
+    }
+}
+
+private extension IdentifyRasterCellView {
+    /// The view model for the sample.
+    @MainActor
+    class Model: ObservableObject {
+        /// A map with an oceans basemap centered on Cape Town, South Africa.
+        let map: Map = {
+            let map = Map(basemapStyle: .arcGISOceans)
+            map.initialViewpoint = Viewpoint(latitude: -34.1, longitude: 18.6, scale: 1_155e3)
+            return map
+        }()
+        
+        /// The raster layer on the map.
+        private var rasterLayer: RasterLayer?
+        
+        /// The placement of the callout on the map.
+        @Published var calloutPlacement: CalloutPlacement?
+        
+        /// The text shown on the callout.
+        @Published private(set) var calloutText: String = ""
+        
+        /// A Boolean value that indicates whether the callout placement should be animated.
+        @Published private(set) var calloutShouldAnimate = false
+        
+        /// A Boolean value that indicates whether to show an error alert.
+        @Published var isShowingErrorAlert = false
+        
+        /// The error shown in the error alert.
+        @Published private(set) var error: Error? {
+            didSet { isShowingErrorAlert = error != nil }
+        }
+        
+        init() {
+            loadRasterLayer()
+        }
+        
+        /// Loads a raster layer from a URL.
+        func loadRasterLayer() {
+            Task { [weak self] in
+                guard let self else { return }
+                
+                do {
+                    // Create a raster with the local file URL.
+                    let raster = Raster(fileURL: .ndviRaster)
+                    
+                    // Create a raster layer using the raster.
+                    rasterLayer = RasterLayer(raster: raster)
+                    
+                    // Load the layer before adding it to the map.
+                    try await rasterLayer?.load()
+                    
+                    // Add the raster layer to the map as an operational layer.
+                    if let rasterLayer {
+                        map.addOperationalLayer(rasterLayer)
+                    } else {
+                        throw CustomError.message("No raster layer to add to the map.")
+                    }
+                } catch {
+                    self.error = error
+                }
+            }
+        }
+        
+        /// Creates a callout displaying the data of a raster cell at a given screen point.
+        /// - Parameters:
+        ///   - screenPoint: The screen point of the raster cell at which to place the callout.
+        ///   - mapViewProxy: The proxy used to handle the screen point.
+        ///   - withOffset: A Boolean value that indicates whether to offset the callout's placement.
+        func callout(at screenPoint: CGPoint, using proxy: MapViewProxy, withOffset: Bool = false) {
+            Task { [weak self] in
```

Comment:
This can also be done using another [task view modifier](https://developer.apple.com/documentation/swiftui/view/task(id:priority:_:)) on the main view instead of kicking off a task here.

So when any of the gestures occur you could save that screen point in a @State variable and then when that value changes you call this method in that task view modifier linked above.